client-spark/common/src/main/java/org/apache/spark/shuffle/ShuffleHandleInfo.java

```diff
@@ -180,6 +195,39 @@ public Map<Integer, List<ShuffleServerInfo>> getPartitionToServers() {
     return partitionToServers;
   }
 
+  /**
+   * key: partitionId, value: the servers for multiple replicas.
+   *
+   * <p>Leveraging the partition reassign mechanism, it could support multiple servers for one
+   * partition replica for huge partition load balance or reassignment multiple times. But it will
+   * use the different policies.
+   *
+   * <p>For the former, this will use the hash to get one from the candidates. For the latter, this
+   * will always get the last one that is available for now.
+   */
+  public Map<Integer, List<ShuffleServerInfo>> getLatestAssignmentPlan(long taskAttemptId) {
```

Comment:
I'd suggest to rename this to
- getAssignmentPlan
- updateAssignmentPlan
- fetchAssignmentPlan

Is `Plan` a thing in uniffle? Is an `AssignmentPlan` something different than an `Assignment`?

This is an assignment for a specific task attempt id. Should this be reflected in the method name?

Reply:
How about `getTaskAssignmentPlan` ? Oh there is no such name of plan, do you think this is reasonable here? I'm not native english and want to hear something from you

Reply:
I am just wondering if `Plan` adds any meaning, or whether `Assignment` is sufficient?

To put it differently, are `Assignment` and `AssignmentPlan` two different concepts or rather two synonyms?

Reply:
After talking with chatgpt, I think `assignmentPlan` is reasonable and it could be used in the english world. like this:

https://ecampusontario.pressbooks.pub/studystrategizesucceed/chapter/develop-an-assignment-plan/

Reply:
An assignment plan is a plan of assignments. What is an assignment, and what is a plan of assignments in uniffle world?

Reply:
> What is an assignment

Assignment is a mapping of partition writing to the corresponding servers.

>  and what is a plan of assignments in uniffle world?

Here I think the assignment plan is the same meaning with the assignment.

Reply:
Then, stick to the simpler term, use Assignment for consistency.

Reply:
Make sense. 

Done.